typescript-test-samples/apigw-lambda-external/README.md

```diff
@@ -0,0 +1,119 @@
+[![typescript: 5.3.2](https://badgen.net/badge/Built%20With/TypeScript/blue9)](https://badgen.net/badge/Built%20With/TypeScript/blue9)
+[![test: unit](https://img.shields.io/badge/Test-Unit-blue)](https://img.shields.io/badge/Test-Unit-blue)
+[![test: integration](https://img.shields.io/badge/Test-Integration-yellow)](https://img.shields.io/badge/Test-Integration-yellow)
+
+# Typescript: Amazon Api Gateway, AWS Lambda, External API dependency
+
+## Introduction
+
+This project contains introductory examples of TypeScript tests written for AWS Lambda. This is a great place to start!
+
+The project uses the [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html) (SAM) CLI for configuration, testing and deployment.
+
+---
+
+## Contents
+
+- [Introduction](#introduction)
+- [Contents](#contents)
+- [About this Pattern](#about-this-pattern)
+- [About this Example](#about-this-example)
+  - [Key Files in the Project](#key-files-in-the-project)
+- [Sample project description](#sample-project-description)
+- [Run the Unit Test](#run-the-unit-test)
+- [Run the Integration Test](#run-the-integration-test)
+
+---
+
+## About this Pattern
+
+### System Under Test (SUT)
+
+The SUT in this pattern is a synchronous API composed of Amazon API Gateway, AWS Lambda and a simple external API.
+
+![System Under Test (SUT)](img/system-under-test.png)
+
+### Goal
+
+The goal of this pattern is to test the SUT in an environment as similar to production as possible by running tests against resources deployed to the cloud but not communicating with an external API.
+
+### Description
+
+The SUT will be deployed to the AWS cloud. The test code will create an HTTP client that will send requests to the deployed API Gateway endpoint. The endpoint will invoke the backing services, test resource configuration, IAM permissions, authorizers, and internal business logic of the SUT. The SUT will not have any dependency on the external API, resulting in the service not being negatively impacted by the constraints or idiosyncrasies of external systems, ensuring that this system remains robust whilst under test.
+
+![System Under Test Description (SUT)](img/system-under-test-description.png)
+
+## [Top](#contents)
+
+## About this Example
+
+This sample project allows a user to call an API endpoint with a city location to retrieve the weather and multi day forecast using an external api [https://goweather.herokuapp.com/weather](https://goweather.herokuapp.com/weather). In this example the AWS API Gateway and AWS Lambda are acting as a proxy and providing no specific business logic over the response from the weather API.
```

Comment:
Consider adjusting to say something like:

"_although_ this example isn't adding additional logic, production systems commonly do work at this layer."

Otherwise, it could just be a straight API Gateway to External API integration with no need for Lambda.